### PR TITLE
Refactor attribute validation

### PR DESCRIPTION
**Pull request by channa-figure**
## Description

Refactor existing Validate method in x/attribute/types/attribute.go to be ValidateBasic to follow sdk standard. 

closes: #44 
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by dpederson-figure — DISMISSED** [review]
I guess this is fine (consistency in naming), but ValidateBasic is a `sdk.Msg`thing, not a general naming convention thing in Go.

File: x/attribute/keeper/keeper.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func (k Keeper) SetAttribute(
 	ctx sdk.Context, acc sdk.AccAddress, attr types.Attribute, owner sdk.AccAddress,
 ) error {
 	// Ensure attribute is valid
-	if err := attr.Validate(); err != nil {
+	if err := attr.ValidateBasic(); err != nil {
```
Comment (iramiller) [inline x/attribute/keeper/keeper.go:107]: The comment removed below (re: stateful check on attribute length) -- this check should be added here in the keeper.

**Review by iramiller — CHANGES_REQUESTED** [review]
We need to add the stateful check on attribute length based on the current configured max value length.  The validation that occurs in the `types` folder must be stateless thus the "validatebasic" refactor...

**Comment by iramiller** [comment]
> I guess this is fine (consistency in naming), but ValidateBasic is a `sdk.Msg`thing, not a general naming convention thing in Go.

Agreed ... the point of this issue originally was to address that the validation done in the attribute type could no longer be performed there due to the stateful property for the attribute length ... this validation has to be done in a keeper method.  I think the point got lost a bit with the `ValidateBasic` focus @dpederson-figure

**Comment by dpederson-figure** [comment]
It's cool. Just a nit.

**Review by iramiller — APPROVED** [review]
Nice, LGTM.

**Review by iramiller — APPROVED** [review]

**Pull request merged**